src/components/UserInfo/UserInfo.tsx

```diff
@@ -0,0 +1,19 @@
+import React from 'react';
+import './UserInfo.scss';
+
+import { User } from '../../types/User';
+
+type Props = {
+  user: User | undefined;
+};
+
+export const UserInfo: React.FC<Props> = ({ user }) => (
+  <div className="user posts__user">
+    <h3 className="user__name">
+      {user && user.name}
+    </h3>
+    <a href={user ? user.email : '#'} className="user__email">
+      {user && user.email}
+    </a>
```

Comment:
```suggestion
    {user && (
      <a href={user.email} className="user__email">
        {user.email}
      </a>
```
Краще використовувати conditional rendering так.
А ще краще це зробити в компоненті PostInfo:
```
{post.user && <UserInfo />}
```